fsck/repair.c

```diff
@@ -92,15 +92,15 @@ bool exfat_repair(struct exfat *exfat, er_problem_code_t prcode,
 	struct exfat_repair_problem *pr = NULL;
 	int need_repair;
 
-	need_repair = ask_repair(exfat, pr);
-	if (!need_repair)
-		return false;
-
 	pr = find_problem(prcode);
 	if (!pr) {
 		exfat_err("unknown problem code. %#x\n", prcode);
 		return false;
 	}
 
+	need_repair = ask_repair(exfat, pr);
+	if (!need_repair)
+		return false;
+
```

Comment:
Why do you move this down?

Reply:
pr is initialized on line #99, while we were already using it on line #95